pulsar-client/src/main/java/org/apache/pulsar/client/api/Reader.java

```diff
@@ -62,4 +62,14 @@
      * Return true if the topic was terminated and this reader has reached the end of the topic
      */
     boolean hasReachedEndOfTopic();
+
+    /**
+     * Check if there is message that has been published successfully to the broker in the topic.
+     */
+    Boolean hasMessageAvailable() throws PulsarClientException;
```

Comment:
To minimize the time "blocked", we could include the last-published message id when the broker sends the `CommandMessage` so that the client can avoid asking, most of the time, since it will already see that the last was ahead of current positition.

Reply:
Yes. we have keep the value as `lastMessageIdInBroker` in `ConsumerImpl`, it will return from client most of the time, will not call into broker.